packages/compiler/core/checker.ts

```diff
@@ -1399,6 +1410,40 @@ export function createChecker(program: Program): Checker {
         );
 
         return undefined;
+      } else if (base.flags & SymbolFlags.MemberContainer) {
+        const type =
+          base.flags & SymbolFlags.LateBound ? base.type! : getTypeForNode(base.declarations[0]);
+        if (
+          type.kind !== "Model" &&
+          type.kind !== "Enum" &&
+          type.kind !== "Interface" &&
+          type.kind !== "Union"
+        ) {
+          program.reportDiagnostic(
+            createDiagnostic({
+              code: "invalid-ref",
+              messageId: "underNamespace",
```

Comment:
Does this message needs to be updated

Reply:
Yeah, there are a few messages to update and asserts that need to be updated/put in place before this gets merged.